libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2170,12 +2170,7 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 		annotation.isLValue = !structType->dataStoredIn(DataLocation::CallData);
 	else if (exprType->category() == Type::Category::Array)
 	{
-		auto const& arrayType(dynamic_cast<ArrayType const&>(*exprType));
-		annotation.isLValue = (
-			memberName == "length" &&
-			arrayType.location() == DataLocation::Storage &&
-			arrayType.isDynamicallySized()
-		);
+		annotation.isLValue = false;
```

Comment:
Could assert inside the if that the member's name is `length`.

Reply:
Wouldn't that potentially break e.g. ``using SomeLibrary for uint256[];`` and then ``uint256[] a; a.someLibraryFunction()``?

Reply:
I'll add a test for that.

Reply:
@ekpyron So, is this the use case you had in mind? It still compiles :-)
```
library Numbers {
    function process(uint256[] storage self) external {}
}
contract TestUnit {
    using Numbers for uint256[];
    uint256[] a;
    function f() public {
        a.process();
    }
}
```

Reply:
Yes, I meant asserting that the name is "length", as @leonardoalt suggested, might have broken that, so better not to assert that.